test/node/advanced.spec.js

```diff
@@ -47,44 +75,99 @@ describe('advanced', () => {
   describe('React', () => {
     describe('element (with circular references)', () => {
       it('compares without warning or errors', () => {
-        const testRenderer = ReactTestRenderer.create(React.createElement(Container));
-        testRenderer.update(React.createElement(Container));
+        const reactApp = ReactTestRenderer.create(
+          React.createElement(ReactContainer)
+        );
+        reactApp.update(React.createElement(ReactContainer));
+        assert.strictEqual(warnStub.callCount, 0);
+      });
+      it('elements of same type and props are equal', () => {
+        const reactApp = ReactTestRenderer.create(
+          React.createElement(ReactContainer)
+        );
+        reactApp.update(React.createElement(ReactContainer));
+        assert.strictEqual(reactChildRenderSpy.callCount, 1);
+      });
+      it('elements of same type with different props are not equal', () => {
+        const reactApp = ReactTestRenderer.create(
+          React.createElement(ReactContainer)
+        );
+        reactApp.update(React.createElement(ReactContainer, { title: 'New' }));
+        assert.strictEqual(reactChildRenderSpy.callCount, 2);
+      });
+    });
+  });
+
+  describe('Preact', () => {
+    let cleanupJsDom;
+
+    before(() => {
+      cleanupJsDom = jsdom();
+    });
+
+    after(() => {
+      PreactTestRenderer.cleanup();
+      cleanupJsDom();
```

Comment:
It's a good practice to never assume anything succeeds -- say `jsdom()` failed, after can still get called IIRC, so might be good to switch to:

```js
if (cleanupJsDom) {
  cleanupJsDom();
}
```